lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -1250,3 +1254,82 @@ def delete_comment(request, comment_id):
         comment_deleted.send(sender=None, user=request.user, post=cc_comment)
     else:
         raise PermissionDenied
+
+
+def _get_course_discussion_user_stats(
+    course_key: CourseKey,
+    order_by: UserOrdering = UserOrdering.BY_FLAGS,
+) -> List[Dict[str, Union[str, int]]]:
+    """
+    Get course user discussion stats.
+
+    Args:
+        course_key (CourseKey): course key for course to fetch discussion stats for.
+        order_by (UserOrdering): the ordering to use for sorting data
+
+    Returns:
+        Sorted user stat data for the specified course
+    """
+    course_stats = get_course_user_stats(course_key)
+
+    def key_func(user):
+        key = (user["threads"], user["responses"], user["replies"])
+        if order_by == UserOrdering.BY_FLAGS:
+            key = (user["active_flags"], user["inactive_flags"]) + key
+        return key
+
+    return sorted(course_stats.values(), key=key_func, reverse=True)
+
+
+def get_course_discussion_user_stats(
```

Comment:
There are fare amount of code in the platform which used NR for capturing function information. `@newrelic.agent.function_trace`
https://docs.newrelic.com/docs/apm/agents/python-agent/python-agent-api/functiontrace-python-agent-api/

How are you planning to capture information on how well the function is performing in prod?

Reply:
@awaisdar001  I was hoping to get feedback on this. I am not familiar with the newrelic API. I will look into this and add the decorators around key functions.

Reply:
thanks.